pkg/workspace/template_loader.go

```diff
@@ -160,7 +160,7 @@ func (l *TemplateLoader) EvalYAML(libraryCtx LibraryExecutionContext, file *file
 	l.ui.Debugf("### ast\n")
 	docSet.Print(l.ui.DebugWriter())
 
-	if !file.IsTemplate() && !file.IsLibrary() {
+	if !(file.IsTemplate() && yamltemplate.IsTemplate(docSet)) && !file.IsLibrary() {
```

Comment:
this is getting hard to understand. should we just add a separate if cond with return instead?

Reply:
No so fast, mister... I think we can do it with a little boolean transformation and a more precise name for this new function...

```go
if !file.IsTemplate() && !file.IsLibrary() || !yamltemplate.HasTemplating(docSet) {
```

huh? huh?